### PR TITLE
Fixes #488

### LINKED ISSUE #488: Please remember "Enable Smart Indent" setting
I was fed up with "Smart Indent" when write HTML.
When I disable "Smart Indent", the trouble gone.
But the option can not save, when restart notepadqq, the option enable again!

### PR DESCRIPTION
**Pull request by qcerasmus**
Added a setting to remember if a user wants smart indentation.

### CODE REVIEW

**Comment by qcerasmus** [comment]
This should also clear up #411 .

**Comment by JuBan1** [comment]
I guess it's a good idea to remember the setting since it does no harm for users who want to disable it only temporarily.

File: src/ui/include/nqqsettings.h [hunk 1]
```diff
@@ -81,6 +81,7 @@ class NqqSettings {
         NQQ_SETTING(RecentDocuments,                QList<QVariant>, QList<QVariant>())
 
         NQQ_SETTING(NotepadqqVersion,               QString,    QString())
+        NQQ_SETTING(SetSmartIndentation,            bool,       true)
```
Comment (danieleds) [inline src/ui/include/nqqsettings.h]: @qcerasmus @JuBan1  wouldn't "SmartIndentation" be better instead of "SetSmartIndentation"?

Comment (JuBan1) [reply src/ui/include/nqqsettings.h]: That came to my mind as well but after looking at the other settings it turns out we actually have a lot of those "verb-style" names. "RememberTabsOnExit", "ShowSpaces", "OverrideFontFamily", "CheckVersionAtStartUp". So it's actually not out of the ordinary.

Comment (danieleds) [reply src/ui/include/nqqsettings.h]: true, however this particular case creates weird/confusing things down in the code:

`m_settings.General.getSetSmartIndentation()`, `m_settings.General.setSetSmartIndentation()`

I think it's better to change it now before it goes into the users settings file, I can do it

**Pull request merged**